Lib/urllib/request.py

```diff
@@ -1494,6 +1494,8 @@ def _is_local_authority(authority):
         if authority == hostname:
             return True
     # Compare IP addresses
+    if not resolve:
```

Comment:
I wonder, should we check this before calling `gethostname()` which can fail?

Reply:
I'm struggling to come up with firm info on when `gethostname()` fails (or whether it can perform network access). Do you have any pointers? Thanks.

Reply:
I do not know.

But I was thinking about this case: processing the "file:" URL on different computer. We cannot use `gethostname()`, because it returns a different result. If we want to get a path on _other_ computer, we should ignore the host name. There will still be problem with processing Windows paths on Posix and Posix paths on Windows.

I do not know how common this case, and how common the opposite case, when we only use `gethostname()`, but not `gethostbyname()`. We should guess what would be more useful for users without having any data.